src/provider/rsa_keymgmt.c

```diff
@@ -257,72 +261,151 @@ static int ibmca_keymgmt_rsa_pub_key_to_data(
     return 1;
 }
 
+static int ibmca_keymgmt_rsa_derive_crt_from_pq(
+                                        const struct ibmca_prov_ctx *provctx,
+                                        BIGNUM *n, BIGNUM *e, BIGNUM *p,
+                                        BIGNUM *q, BIGNUM *dp, BIGNUM *dq,
+                                        BIGNUM *qinv)
+{
+    BIGNUM *d = NULL, *p1 = NULL, *q1 = NULL;
+    BN_CTX *ctx = NULL;
+    int rc = 0;
+
+    ctx = BN_CTX_secure_new();
+    if (ctx == NULL) {
+        put_error_ctx(provctx, IBMCA_ERR_MALLOC_FAILED,
+                      "BN_CTX_secure_new failed");
+        goto out;
+    }
+
+    p1 = BN_CTX_get(ctx);
+    q1 = BN_CTX_get(ctx);
+    if (p1 == NULL || q1 == NULL) {
+        put_error_ctx(provctx, IBMCA_ERR_MALLOC_FAILED, "BN_CTX_get failed");
+        goto out;
+    }
+
+    BN_set_flags(p1, BN_FLG_CONSTTIME);
+    BN_set_flags(q1, BN_FLG_CONSTTIME);
+
+    if (!BN_sub(p1, p, BN_value_one()) || /* p-1 */
+        !BN_sub(q1, q, BN_value_one())) { /* q-1 */
+        put_error_ctx(provctx, IBMCA_ERR_MALLOC_FAILED, "BN_sub failed");
+        goto out;
+    }
+
+    BN_set_flags(dp, BN_FLG_CONSTTIME);
+    BN_set_flags(dq, BN_FLG_CONSTTIME);
+    BN_set_flags(qinv, BN_FLG_CONSTTIME);
+
+    rc = ibmca_keymgmt_rsa_calc_priv_d(provctx, n, e, p, q, &d);
+    if (rc != 1)
+        goto out;
+
+    rc = 0;
+
+    /* (Step 5a) dP = d mod (p-1) */
```

Comment:
Which algorithm is referred to here with these steps? Maybe provide a short info or URL?

Reply:
fixed